subsys/net/lib/coap/coap_client.c

```diff
@@ -27,6 +27,7 @@ static atomic_t coap_client_recv_active;
 static int send_request(int sock, const void *buf, size_t len, int flags,
```

Comment:
While we are in the area, can we please add some comments to `send_request`, `send_ack` and `handle_request` indicating the intended expected return values?

For `send_request` and `send_ack` it is pretty simple, they return a negative value on error, and a positive value indicating total number of bytes sent on success.


But `handle_request` is very difficult to untangle due to its inconsistent and undocumented control flow. It actually has three different non-negative return values (0, 1, and under certain conditions, the exact value that `send_ack` returned, verbatim), which are inconsistent enough as to not even be useful to a potential caller. And the `fail` block is, IMO, misleadingly named.

Therefore, could we additionally please:
1. Make those return values consistent
2. Add a comment to the `send_ack` control flow indicating that the passthrough into the `fail` block is ***intentional***, and is (I presume) done because sending an ack indicates that the request has completed (if it is non-OBSERVE)
3. Add a comment to the `send_request` control flow indicating that we return early with `1` because (I presume) the overall request is still ongoing due to the followup request?
4. Possibly rename the `fail` block? -- That is, if you can think of a better name. Maybe just `exit` or `cleanup` or `conclude` or `finish`? IDK... Since this block now also handles OBSERVE requests, its kinda difficult to name properly. Maybe this indicates the function should be broken down differently, but I won't ask you to totally refactor it 😅

Reply:
Discussed offline, these readability tweaks are out of scope for what we are working on. But @tautologyclub you may consider them my recommendations to you for when you do that followup PR you mentioned

Reply:
That being said, @juhaylinen What do you think of these suggestions? Shall I file a github issue about them?